### ISSUE TITLE
NoneType != support

### ISSUE THREAD

**Issue by ge-smile** [serge-sans-paille/pythran#1206]
Hi, 

We are missing "!=" operator overloaded in pythran/pythran/pythonic/types/NoneType.hpp. 

Also for "==" operator for NoneType, we seem not consider the case when O is None type. The function I mean is:

  template<class T>
  template<class O>
  bool none<T, false>::operator==(O const &t) const
  {
    return !is_none && static_cast<const T &>(*this) == t;
  }

Thanks,
Alice

**Comment by serge-sans-paille** [comment]
Thanks for reporting! I'll have a look.

**Comment by serge-sans-paille** [comment]
Concerning case when O is none, there is an overload

```
  template <class T>
  bool none<T, false>::operator==(none_type const &) const
  {
    return is_none;
  }
```

that should handle it.

I confirm that `operator!=` is missing, i'm adding it.

**Comment by serge-sans-paille** [comment]
Fixed by #1207

**Comment by ge-smile** [comment]
Thanks for the fix!
